### PR TITLE
Show/hide maintenance devices

### PR DESCRIPTION
**Pull request by gdepeyrot**
Added WorldMap possibility to show/hide devices under maintenance

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Http/Controllers/Widgets/WorldMapController.php [hunk 1]
```diff
@@ -44,7 +44,8 @@ public function __construct()
             'init_lng' => Config::get('leaflet.default_lng', 0),
             'init_zoom' => Config::get('leaflet.default_zoom', 2),
             'group_radius' => Config::get('leaflet.group_radius', 80),
-            'status' => '0,1',
+            'status' => '0,1', // Show all devices
+            'maintenance' => '0', // Hide devices under maintenance
```
Comment (murrant) [inline app/Http/Controllers/Widgets/WorldMapController.php]: Any reason you are using textual numbers instead of just numbers?

Comment (gdepeyrot) [reply app/Http/Controllers/Widgets/WorldMapController.php]: No, I guess not.

**Comment by Jellyfrog** [comment]
The logic should be moved to the sql query above, so you only retrieve the devices you want, then you don't need the php logic

**Comment by gdepeyrot** [comment]
> The logic should be moved to the sql query above, so you only retrieve the devices you want, then you don't need the php logic

The sql query above already filters the device according to the selected status (Up + Down, Up or Down). AFAIU, there is no simple query to filter devices under maintenance. Is it possible to combine the isUnderMaintenance check into the same query?

Maybe I haven't understood what you are suggesting.

**Comment by Jellyfrog** [comment]
> > The logic should be moved to the sql query above, so you only retrieve the devices you want, then you don't need the php logic
> 
> The sql query above already filters the device according to the selected status (Up + Down, Up or Down). AFAIU, there is no simple query to filter devices under maintenance. Is it possible to combine the isUnderMaintenance check into the same query?
> 
> Maybe I haven't understood what you are suggesting.

You're right, I forgot it comes from another table.
It is possible to make it a "scope function" like it's called in laravel, do you think it's worth it @murrant ?

**Comment by PipoCanaja** [comment]
@murrant @Jellyfrog I let you continue on this one. From my side, looks good, but I am not the DB guy at all :)

**Comment by murrant** [comment]
You should be able to create a scope on the Devices model by using the isUnderMaintenance query as a subquery.

**Comment by gdepeyrot** [comment]
Ok, I'll have a look at how to implement that.

**Comment by kkrumm1** [comment]
Any update on this?

**Comment by gdepeyrot** [comment]
Unfortunately not. Been swamped with other subjects and have not been able to look into this.

**Comment by SourceDoctor** [comment]
@gdepeyrot 
something new about pending changes?

**Comment by gdepeyrot** [comment]
No, nothing new. Don't know when I'll have time to learn how to create the suggested scope. Right now concentrated on the survival of my company during the COVID-19 crisis...

**Comment by Jellyfrog** [comment]
Closing due to inactivity, feel free to reopen

**Pull request closed without merging**